appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java

```diff
@@ -43,7 +47,32 @@ public NewYoungAndroidProjectWizard() {
     // Initialize the UI
     setStylePrimaryName("ode-DialogBox");
 
-    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel());
+    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
+      @Override
+      public boolean validate(String value) {
+        if(!value.matches("[A-Za-z][A-Za-z0-9_]*") && value.length() > 0) {
```

Comment:
see

http://stackoverflow.com/questions/10894122/java-regex-for-support-unicode

or do a web search under

java unicode regexp



== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty


On Tue, Jun 16, 2015 at 6:57 PM, KeithGalli <notifications@github.com>
wrote:

> In
> appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java
> <https://github.com/mit-cml/appinventor-sources/pull/402#discussion_r32580649>
> :
>
> > @@ -43,7 +47,32 @@ public NewYoungAndroidProjectWizard() {
> >      // Initialize the UI
> >      setStylePrimaryName("ode-DialogBox");
> >
> > -    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel());
> > +    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
> > +      @Override
> > +      public boolean validate(String value) {
> > +        if(!value.matches("[A-Za-z][A-Za-z0-9_]*") && value.length() > 0) {
>
> @jisqyv <https://github.com/jisqyv> Any ideas on how to get the unicode
> regex expressions to work with app inventor?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/402/files#r32580649>.
>